src/annotator/anchoring/test/html-test.js

```diff
@@ -419,42 +422,133 @@ describe('HTML anchoring', function () {
   describe('Web page baselines', function () {
     let frame;
 
-    before(function () {
+    beforeEach(() => {
       frame = document.createElement('iframe');
       document.body.appendChild(frame);
     });
 
-    after(function () {
+    afterEach(() => {
       frame.remove();
+      $imports.$restore();
     });
 
     htmlBaselines.forEach(fixture => {
-      it(`generates selectors which match the baseline ${fixture.name}`, () => {
-        let fixtureHtml = fixture.html;
-        const annotations = fixture.annotations.rows;
-
-        // Break references to external stylesheets and images so that the test
-        // runner does not try to load them.
-        //
-        // This method is crude but Good Enough™ for our test fixtures and does
-        // not modify the offset of any substrings within the text content of the
-        // document for matches that occur in a text-content context (eg.
-        // <pre>, <noscript>).
-        fixtureHtml = fixtureHtml
-          .replace(/src=/g, 'sr_=')
-          .replace(/href=/g, 'hre_=');
-
-        frame.contentWindow.document.documentElement.innerHTML = fixtureHtml;
-
-        const annotationsChecked = annotations.map(async ann => {
-          const root = frame.contentWindow.document.body;
-          const selectors = ann.target[0].selector;
-          const range = await html.anchor(root, selectors);
-          const newSelectors = await html.describe(root, range);
-          assert.deepEqual(sortByType(selectors), sortByType(newSelectors));
+      [
+        {
+          name: 'when no anchors throw an error',
```

Comment:
It is good to add test cases to cover the current gaps in this module's tests, but I don't think they should be added to "Web baselines" tests. These tests are intended to test anchoring with real webpages and selectors. As a result they are more complex to debug if something fails.

Instead I think it would be better to test this fallback behavior using tests more like the "when anchoring fails" test block above.